src/main/java/com/codeborne/selenide/impl/ScreenShotLaboratory.java

```diff
@@ -332,12 +341,37 @@ public void startContext(String context) {
     }
   }
 
+  public List<File> getThreadScreenshots() {
+    return Collections.unmodifiableList(threadScreenshots.get());
+  }
+
+  public List<File> getContextScreenshots() {
+    List<File> screenshots = currentContextScreenshots.get();
+    return Collections.unmodifiableList(screenshots == null
+      ? Collections.emptyList()
+      : screenshots);
+  }
+
   public File getLastScreenshot() {
     synchronized (allScreenshots) {
       return allScreenshots.isEmpty() ? null : allScreenshots.get(allScreenshots.size() - 1);
     }
   }
 
+  public File getLastThreadScreenshot() {
+    List<File> screenshots = threadScreenshots.get();
+    return screenshots.isEmpty()
+      ? null
```

Comment:
probably return Optional instead of null?

Reply:
yes, initially I thought to return Optional, but then decided to make it similar to existing getScreenthods()
but I agree, let's return Optional